### PR TITLE
Add documentation for hyperparameter importance target parameter

### PR DESCRIPTION
**Pull request by Crissman**
## Motivation
Hyperparameter importance (HPI) can use a callable function to look at the importance of other parameters, but many users may not be familiar with how to do this.

## Description of the changes
Update the hyperparameter and visualization tutorials to highlight this ability. Likewise, update the docs and error messages for HPI to show the same.

### CODE REVIEW

File: optuna/importance/_base.py [hunk 1]
```diff
@@ -44,11 +44,13 @@ def evaluate(
...
                 If it is :obj:`None` and ``study`` is being used for single-objective optimization,
-                the objective values are used.
+                the objective values are used. Can also be used for other trial parameters, such as
+                the duration, like `target=lambda t: t.duration.total_seconds()`.
 
                 .. note::
                     Specify this argument if ``study`` is being used for multi-objective
-                    optimization.
+                    optimization. For example, to get the hyperparameter importance of the first
+                    objective, use `target=lambda t: t.value[0]` for the target parameter.
```
Comment (nzw0301) [inline optuna/importance/_base.py:53]:
```suggestion
                    objective, use `target=lambda t: t.values[0]` for the target parameter.
```

File: optuna/importance/_fanova/_evaluator.py [hunk 2]
```diff
@@ -80,7 +80,8 @@ def evaluate(
         if target is None and study._is_multi_objective():
             raise ValueError(
                 "If the `study` is being used for multi-objective optimization, "
-                "please specify the `target`."
+                "please specify the `target`. For example, use "
+                "`target=lambda t: t.value[0]` for the first objective value."
```
Comment (nzw0301) [inline optuna/importance/_fanova/_evaluator.py:84]:
```suggestion
                "`target=lambda t: t.values[0]` for the first objective value."
```

File: optuna/importance/_mean_decrease_impurity.py [hunk 3]
```diff
@@ -63,7 +63,8 @@ def evaluate(
         if target is None and study._is_multi_objective():
             raise ValueError(
                 "If the `study` is being used for multi-objective optimization, "
-                "please specify the `target`."
+                "please specify the `target`. For example, use "
+                "`target=lambda t: t.value[0]` for the first objective value."
```
Comment (nzw0301) [inline optuna/importance/_mean_decrease_impurity.py:67]:
```suggestion
                "`target=lambda t: t.values[0]` for the first objective value."
```

File: tutorial/20_recipes/002_multi_objective.py [hunk 4]
```diff
@@ -121,3 +121,10 @@ def objective(trial):
 ###################################################################################################
 # Check trials on pareto front visually
 optuna.visualization.plot_pareto_front(study, target_names=["FLOPS", "accuracy"])
+
+
+###################################################################################################
+# Learn which hyperparameters are affecting the flops most with hyperparameter importance
+optuna.visualization.plot_param_importances(
+    study, target=lambda t: t.value[0], target_name="flops"
```
Comment (nzw0301) [inline tutorial/20_recipes/002_multi_objective.py:129]:
```suggestion
    study, target=lambda t: t.values[0], target_name="flops"
```

**Review by nzw0301 — CHANGES_REQUESTED** [review]

**Comment by nzw0301** [comment]
Thank you for your explanation! I felt the exact same thing for `plot_optimization_history` with multi-objective functions. I leave minor comments and one of them will fix the CI error related to the document.

**Comment by toshihikoyanase** [comment]
@nzw0301 Could you continue the review, please?

**Review by nzw0301 — APPROVED** [review]
LGTM!

**Comment by nzw0301** [comment]
Sorry for my careless review. How about adding the description to the arguments of both `optuna.visualization.plot_param_importances` and `optuna.visualization.matplotlib.plot_param_importances` ?

**Comment by Crissman** [comment]
Thanks, @nzw0301  Caught a few other places that it should be included. 👍

**Comment by nzw0301** [comment]
Thanks for your quick update! 

I have a minor comment about backquote. The current `target=lambda t:  ...` is surrounded by a single backquote on both sides, so the documentation looks 

Screenshot 2021-04-02 at 18 00 57

If we add additonal backquote to each side:

## if we add additional backquoto
Screenshot 2021-04-02 at 18 00 38

**Comment by Crissman** [comment]
Doubled down on quotes.

**Review by nzw0301 — APPROVED** [review]
Thanks!

File: optuna/importance/_base.py [hunk 5]
```diff
@@ -44,11 +44,13 @@ def evaluate(
             target:
                 A function to specify the value to evaluate importances.
                 If it is :obj:`None` and ``study`` is being used for single-objective optimization,
-                the objective values are used.
+                the objective values are used. Can also be used for other trial parameters, such as
```
Comment (toshihikoyanase) [inline optuna/importance/_base.py:47]: I found `trial parameters` a bit confusing. Readers might recognize it as `trial.parameters`. How about rephrasing it?
```suggestion
                the objective values are used. Can also be used for other trial attributes, such as
```

File: tutorial/10_key_features/005_visualization.py [hunk 6]
```diff
@@ -106,6 +106,12 @@ def objective(trial):
 # Visualize parameter importances. See :func:`~optuna.visualization.plot_param_importances` for the details.
 plot_param_importances(study)
 
+###################################################################################################
+# Learn which hyperparameters are affecting the trial duration with hyperparameter importance
```
Comment (toshihikoyanase) [inline tutorial/10_key_features/005_visualization.py:110]: A nit, but other texts have commas at the end of lines.
```suggestion
# Learn which hyperparameters are affecting the trial duration with hyperparameter importance.
```

File: tutorial/20_recipes/002_multi_objective.py [hunk 7]
```diff
@@ -121,3 +121,10 @@ def objective(trial):
 ###################################################################################################
 # Check trials on pareto front visually
 optuna.visualization.plot_pareto_front(study, target_names=["FLOPS", "accuracy"])
+
+
+###################################################################################################
+# Learn which hyperparameters are affecting the flops most with hyperparameter importance
```
Comment (toshihikoyanase) [inline tutorial/20_recipes/002_multi_objective.py:127]: ditto.
```suggestion
# Learn which hyperparameters are affecting the flops most with hyperparameter importance.
```

**Review by toshihikoyanase — COMMENTED** [review]
Thank you for your PR. I added small comments.

**Review by toshihikoyanase — APPROVED** [review]
Thank you for your update. LGTM!

**Pull request merged**